src/edit/main.js

```diff
@@ -156,11 +156,22 @@ class ProseMirror {
       domDrop: new DOMSubscription
     }
 
-    if (opts.place && opts.place.appendChild)
+    if (opts.place && opts.place.appendChild) {
+      this.root = opts.place
+
+      // loop if root is not document (in light dom)
+      // only shadowRoot has property (in shadow dom)
+      while (this.root!== document && !this.root.host) {
```

Comment:
> it could be async

Yep fair call. The docs don't specify whether it must be placed sync, so it's reasonable to assume async is acceptable.

> Determines the placement of the editor in the page. When null, the editor is not placed. When a DOM node is given, the editor is appended to that node. When a function is given, it is called with the editor's wrapping DOM node, and is expected to place it into the document.

It's unfortunate that `Function` is a valid value for `place`.